server/routes/activities/create-an-activity/handlers/sessionTimesOption.ts

```diff
@@ -31,8 +31,21 @@ export default class SessionTimesOptionRoutes {
 
   POST = async (req: Request, res: Response): Promise<void> => {
     const { usePrisonRegimeTime } = req.body
+    const { preserveHistory } = req.query
+
     if (usePrisonRegimeTime === 'true') {
-      res.redirectOrReturn('bank-holiday-option')
-    } else res.redirectOrReturn(`session-times`)
+      req.session.createJourney.customSlots = undefined
+      if (preserveHistory === 'true') {
+        return res.redirect('check-answers')
+      }
+      return res.redirectOrReturn('bank-holiday-option')
+    }
+
+    let redirectParams = ''
+    if (preserveHistory === 'true') {
+      redirectParams += '?preserveHistory=true'
+    }
+
+    return res.redirect(`session-times${redirectParams}`)
```

Comment:
If we are coming from check answers and user select custom then go to session times page